Mage.Sets/src/mage/cards/s/SailIntoTheWest.java

```diff
@@ -0,0 +1,149 @@
+                "If embark gets more votes or the vote is tied, each player may discard their hand and draw seven cards.";
+    }
+
+    private SailIntoTheWestEffect(final SailIntoTheWestEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public SailIntoTheWestEffect copy() {
+        return new SailIntoTheWestEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        TwoChoiceVote vote = new TwoChoiceVote(
+            "Return (return up to 2 cards from graveyard)",
+            "Embark (may discard all and draw 7)",
+            Outcome.Benefit);
+        vote.doVotes(source, game);
+
+        int returnCount = vote.getVoteCount(true);
+        int embarkCount = vote.getVoteCount(false);
+
+        if(returnCount > embarkCount) {
+            return applyReturn(game, source);
+        }
+        else {
+            return applyEmbark(game, source);
+        }
+    }
+
+    private boolean applyReturn(Game game, Ability source) {
+        game.informPlayers("'Return' won the vote. Each player may return up to two cards from their graveyard to their hand");
+        for (UUID playerId : game.getState().getPlayersInRange(source.getControllerId(), game)) {
+            Player player = game.getPlayer(playerId);
+
+            if (player != null) {
+                FilterCard filter = new FilterCard("card from your graveyard");
+                filter.add(new OwnerIdPredicate(player.getId()));
+                Target targetCard = new TargetCardInGraveyard(0,2,filter);
+                targetCard.setNotTarget(true);
+
+                if (targetCard.canChoose(player.getId(), source, game)) {
+                    if (player.chooseTarget(Outcome.ReturnToHand, targetCard, source, game)) {
+                        Set<Card> cards = targetCard
+                            .getTargets()
+                            .stream()
+                            .map(game::getCard)
+                            .collect(Collectors.toSet());
+                        if (!cards.isEmpty()) {
+                            player.moveCards(cards, Zone.HAND, source, game);
+                        }
+                        else {
+                            game.informPlayers(player.getLogName() + " decided to not return anything.");
+                        }
+                    }
+                }
+            }
+        }
+        return true;
+    }
+
+    private boolean applyEmbark(Game game, Ability source) {
+        game.informPlayers("'Embark' won the vote. Each player may discard their hand and draw seven cards.");
+
+        for (UUID playerId : game.getState().getPlayersInRange(source.getControllerId(), game)) {
+            Player player = game.getPlayer(playerId);
+
+            if (player != null) {
```

Comment:
Should that be `chooseUse(Outcome.DrawCard, "discard your hand and draw seven cards?", source, game)` instead?
Both should work, but not the same choice UI ?

Reply:
The four-argument constructor calls the seven-argument constructor with default `null`, `"Yes"`, `"No"`. Simpler should be fine here but not a big deal either way.